lms/djangoapps/grades/tasks.py

```diff
@@ -87,12 +104,24 @@ def _has_database_updated_with_new_score(
         # score should be None only if it was deleted.
         # Otherwise, it hasn't yet been saved.
         return score_deleted
+    elif score.grade is None:  # submissions-related problem
```

Comment:
Can you check the `module_type` instead?  I'm not sure that checking for `grade` will always work.